src/ingestion/sources/redshift.py

```diff
@@ -0,0 +1,59 @@
+import json
+
+from .base import SQLAlchemyExtractor, SourceConfiguration, SQLAlchemySourceDialect, SQLAlchemySource
+from .postgresql import PostgreSQLSourceDialect
+
+class RedshiftSourceConfiguration(SourceConfiguration):
+    @classmethod
+    def validate(cls, configuration):
+        raise NotImplementedError
+
+    @classmethod
+    def configuration_schema(cls):
+        return {
+            "type": "object",
+            "properties": {
+                "user": { "type": "string" },
+                "password": { "type": "string" },
+                "host": { "type": "string" },
+                "port": { "type": "string" },
+                "database": { "type": "string" },
+                "schema": { "type": "string" },
+            },
+            "secret": [ "password" ],
+        }
+
+    def connection_string(self) -> str:
+        configuration = json.loads(self.configuration)
+
+        return 'redshift+psycopg2://{user}:{password}@{host}:{port}/{database}'.format(
+            user=configuration.get('user'),
+            password=configuration.get('password'),
+            host=configuration.get('host'),
+            port=configuration.get('port'),
+            database=configuration.get('database'),
+            schema=configuration.get('schema'),
+        )
+
+    @property
+    def type(self):
+        return "redshift"
+
+class RedshiftSourceDialect(PostgreSQLSourceDialect):
+    pass
+
+class RedshiftSourceExtractor(SQLAlchemyExtractor):
+    def discovery_query(self):
+        return RedshiftSourceDialect.schema_columns_query(
+            database_name=self.configuration.database(),
+            schema_name=self.configuration.schema(),
+        )
+
+class RedshiftSource(SQLAlchemySource):
+    def __init__(self, configuration: RedshiftSourceConfiguration):
+        self.configuration = configuration
+        self.dialect = RedshiftSourceDialect
+
+    def extractor(self):
```

Comment:
If no difference in dialect, don't need this extractor or redshiftsourcedialect at all, just subclass pg